### ISSUE TITLE
Refactor live "/:username", ProfileLive

### ISSUE THREAD

**Issue by wintermeyer** [wintermeyer/animina#1004]
Feed the database with the demo users and load a `/:username` page in your browser. Have a look at the millions of SQL queries which are triggered by that page.

I am sure that there is a massive potential to DRY that code and to improve it. No need to load the same data over and over again from the database.

@briankariuki Please do create sub issues when it makes sense. I prefer small and quick over big PRs since everybody else is working on the same code.

PS: Please add a before/after time when possible. I would like to get a feeling how much performance we gain with this issue.

## Tasks
- [x] #1030
- [x] #1031
- [x] #1033
- [x] #1028
- [x] #1049
- [x] #1056
- [x] #1054
- [ ] Refactor profile_live

**Comment by wintermeyer** [comment]
Code readability is a major point too. Since the application is online now more people look into the code. They should have a chance to understand it.

**Comment by briankariuki** [comment]
Let me investigate what's causing the issue. I'll add the findings here before attempting to fix it

**Comment by briankariuki** [comment]
The first issue I've noticed is the read [:by_user_id](https://github.com/animina-dating/animina/blob/ca8253b390eaf94089df1330734d1cbc77e5420e/lib/animina/narratives/resources/story.ex#L69) action on the Story resource. 
It loads the user, which is redundant as already the user_id is passed. The User resource [loads](https://github.com/animina-dating/animina/blob/ca8253b390eaf94089df1330734d1cbc77e5420e/lib/animina/accounts/resources/user.ex#L375) a few things when its built which require reading from the database. Hence the many sql query reads for each story fetched with the user preloaded.

I can go ahead and refactor the Story resource not to load the user or do it on the [profile_stories_live.ex](https://github.com/animina-dating/animina/blob/ca8253b390eaf94089df1330734d1cbc77e5420e/lib/animina_web/live/profile_stories_live.ex#L235) where we fetch the stories and use the :read action and preload only what we need to display that page.

@wintermeyer  what do you think?

**Comment by briankariuki** [comment]
The second issue I found is that the [UserFlags](https://github.com/animina-dating/animina/blob/ca8253b390eaf94089df1330734d1cbc77e5420e/lib/animina/traits/resources/user_flags.ex#L71) resource also preloads the user.

**Comment by wintermeyer** [comment]
> @wintermeyer what do you think?

I can't say much about it because I am not deep in the code. I just see all the SQL queries in the log and always think "this can not be the optimal way of doing it". Within the Phoenix world SQL queries are expensive (at least 9 out of 10 times). I am saying this because I used to program a lot with Ruby on Rails and there SQL was much cheaper than Ruby. But for us normal Elixir is cheaper than multiple SQL queries.

Bottom line: Do what makes most sense for you to get more speed and improve readability (hoping that both come together). In case you have to make a choice between speed and code readability I tend to go into the code readability direction.

**Comment by wintermeyer** [comment]
In my opinion the `/:username` ProfileLive should not need more than 2-4 SQL queries. In a perfect world it should only need 1 but let's be realistic.

**Comment by wintermeyer** [comment]
> The second issue I found is that the [UserFlags](https://github.com/animina-dating/animina/blob/ca8253b390eaf94089df1330734d1cbc77e5420e/lib/animina/traits/resources/user_flags.ex#L71) resource also preloads the user.

Please keep #1002 in mind while working with the flags.

**Comment by wintermeyer** [comment]
@briankariuki Please do me a favour and set the `in progress` flag to this issue while you are working on it. Thanks!
